toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -200,8 +215,49 @@ func validateScript(baseConfigPath string, script *imagecustomizerapi.Script) er
 	return nil
 }
 
+func validatePackageLists(baseConfigPath string, config *imagecustomizerapi.SystemConfig, rpmsSources []string,
+	useBaseImageRpmRepos bool, partitionsCustomized bool,
+) error {
+	allPackagesRemove, err := collectPackagesList(baseConfigPath, config.PackageListsRemove, config.PackagesRemove)
+	if err != nil {
+		return err
+	}
+
+	allPackagesInstall, err := collectPackagesList(baseConfigPath, config.PackageListsInstall, config.PackagesInstall)
+	if err != nil {
+		return err
+	}
+
+	allPackagesUpdate, err := collectPackagesList(baseConfigPath, config.PackageListsUpdate, config.PackagesUpdate)
+	if err != nil {
+		return err
+	}
+
+	hasRpmSources := len(rpmsSources) > 0 || useBaseImageRpmRepos
```

Comment:
is `rpmsSources` inferred or passed from user?

Reply:
Then I'm confused by the if condition `if !hasRpmSources`. 
If you have custom partitions in addition to packages to install and relying on pmc. Then this condition will be `true`, but then the following errors will be hit. Am i missing something?

Reply:
If `--disable-base-image-rpm-repos` isn't set, then `useBaseImageRpmRepos` will be `true` and so `hasRpmSources` will be `true` and so `!hasRpmSources` will be false and so the errors won't trigger.